NativeApp/src/components/Booking/container.js

```diff
@@ -138,6 +137,25 @@ export default Container => class extends Component {
       ));
   }
 
+  cancelHoliday = () => {
+    const { booking } = this.state;
+    const { holId } = booking;
+    const { navigation } = this.props;
+
+    const request = {
+      holidayId: holId,
+    };
+
+    cancelHolidayRequest(request)
+      .then(() => {
```

Comment:
I think (if you wanted) you could have `.then(navigation.pop)` here

Reply:
Would it not get called before? Like it was on login.

Reply:
Nope, so

`.then(navigation.pop)` is passing along a reference to the pop function within the navigation object 

`.then(() => navigation.pop())` is passing along an anonymous function which will call the pop function within the navigation object. This is more or less the same as above (I think), only slightly less cleaner.

`.then(navigation.pop())` would immediately call the pop function within the navigation object, and pass whatever it gets back to `then`, which isn't what we want here.

Hope that clears it up a bit!

Reply:
oooo got it now. I didn't see the `()` that was missing from what you suggested.

Reply:
Actually just tried it. Doesn't fire the `pop` action.

Reply:
@bgribz Had to revert it back. Seems `() => navigation.pop()` is the only way it will work.